text_2_sql/text_2_sql_core/src/text_2_sql_core/prompts/query_rewrite_agent.yaml

```diff
@@ -1,56 +1,124 @@
+         - Look for superlatives, multiple dimensions, or comparisons
+         - Determine if breaking down would simplify processing
+
+      2. Break Down Complex Queries:
+         - Create independent sub-queries that can be processed separately
+         - Ensure each sub-query is simple and focused
+         - Include clear combination instructions
+         - Preserve all necessary context in each sub-query
+
+      3. Handle Date References:
+         - Resolve relative dates using {{ current_datetime }}
+         - Maintain consistent YYYY-MM-DD format
+         - Include date context in each sub-query
+
+      4. Maintain Query Context:
+         - Each sub-query should be self-contained
+         - Include all necessary filtering conditions
+         - Preserve business context
 
     <rules>
-        1. **Understanding**: Use the chat history (that is available in reverse order) to understand the context of the current question. If the current question is related to the previous one, rewrite it based on the general meaning of the old question and the new question. If they do not relate, output the new question as is.
-        2. **Date Resolution Second**: Resolve all relative dates before decomposing questions.
-        3. **Self-Contained Questions**: Ensure each decomposed question is independent and includes all necessary context, without referencing the original question.
-        4. **Consistency**: Maintain a uniform structure across all rewritten questions.
-        5. **Simplification**: If the question is already simple but includes relative dates, resolve the dates without decomposition.
+        1. Always consider if a complex query can be broken down
+        2. Make sub-queries as simple as possible
+        3. Include clear instructions for combining results
+        4. Preserve all necessary context in each sub-query
+        5. Resolve any relative dates before decomposition
     </rules>
 
     <output_format>
-        Return an array of rewritten questions as valid JSON:
-        - For decomposed questions:
-          [\"<rewritten_question_1>\", \"<rewritten_question_2>\"]
-
-        - For simple questions (date resolution only):
-          [\"<rewritten_question>\"]
+        Return a JSON object with sub-queries and combination instructions:
+        {
+          "sub_queries": [
+            "<sub_query_1>",
+            "<sub_query_2>",
+            ...
+          ],
+          "combination_logic": "<instructions for combining results>",
+          "query_type": "<simple|complex>"
+        }
     </output_format>
   </instructions>
 
   <examples>
-    - **Input**: \"How much did we make in sales last month and what were our top products?\"
-      **Output**:
-      [\"How much did we make in sales in November 2024?\", \"What were our top products in November 2024?\"]
+    Example 1:
+    Input: "What country did we sell the most to in June 2008?"
```

Comment:
Maybe we should get different examples here? Perhaps these questions don't need to be decomposed as they are answerable in one query?